test/controllers/api/v2/activation_keys_controller_test.rb

```diff
@@ -116,6 +224,58 @@ def test_update
       assert_template 'api/v2/activation_keys/show'
     end
 
+    test_attributes :pid => '0f857d2f-81ed-4b8b-b26e-34b4f294edbc'
+    def test_should_not_update_with_invalid_max_hosts
+      put :update, params: {
+        :id => @activation_key.id,
+        :organization_id => @organization.id,
+        :activation_key => { :unlimited_hosts => false, :max_hosts => 0 }
+      }
+      assert_response :unprocessable_entity
+      assert_match 'Validation failed: Max hosts cannot be less than one', @response.body
+    end
+
+    test_attributes :pid => 'da85a32c-942b-4ab8-a133-36b028208c4d'
+    def test_should_not_update_with_invalid_name
+      put :update, params: {
+        :id => @activation_key.id,
+        :organization_id => @organization.id,
+        :activation_key => { :unlimited_hosts => false, :max_hosts => 0 }
+      }
+      assert_response :unprocessable_entity
+      assert_match 'Validation failed: Max hosts cannot be less than one', @response.body
+    end
+
+    test_attributes :pid => '3bcff792-105a-4577-b7c2-5b0de4f79c77'
+    def test_should_not_update_existing_with_invalid_max_hosts
+      activation_key = ActivationKey.new(
+        :name => 'new key', :organization => @organization, :unlimited_hosts => false, :max_hosts => 1
+      )
+      assert activation_key.save
+      put :update, params: {
+        :id => activation_key.id,
+        :organization_id => @organization.id,
+        :activation_key => { :max_hosts => 'foo' }
+      }
+      assert_response :unprocessable_entity
+      assert_match 'Validation failed: Max hosts is not a number', @response.body
+    end
+
+    test_attributes :pid => 'ec225dad-2d27-4b37-989d-1ba2c7f74ac4'
```

Comment:
can you explain what these test_attributes do ?

Reply:
@jlsherrill test_attributes is declared here https://github.com/SatelliteQE/robottelo_reporter/blob/master/lib/robottelo/reporter/attributes.rb
The main purpose of this call is to register the latest values of some key names, after this call (as after this function call we have a method declaration)  method_added(meth) will be called, that will found that there is some latest attributes and will store that method name with that keys values 

Later when the report plugin method are called we can retrieve attributes information about that tests using get_test_attributes method.

Reply:
@jlsherrill please also refer to rebottelo-reporter usage: https://github.com/SatelliteQE/robottelo_reporter#usage